profiles/views/quota_attribute_definition_views.py

```diff
@@ -0,0 +1,90 @@
+        context['html_button_path'] = "generics/buttons/generic_add_button.html"
+        context['app_name'] = "profiles"
+        context['object_name'] = "quota_attribute_definition"
+        return context
+
+
+@user_passes_test(lambda u: u.is_superuser)
+def quota_attribute_definition_edit(request, quota_attribute_definition_id):
+    quota_attribute_definition = get_object_or_404(QuotaAttributeDefinition, id=quota_attribute_definition_id)
+    form = QuotaAttributeDefinitionForm(request.POST or None, instance=quota_attribute_definition)
+    if form.is_valid():
+        form.save()
+        return redirect("profiles:quota_attribute_definition_list")
+    breadcrumbs = [
+        {'text': 'Quota attributes', 'url': reverse('profiles:quota_attribute_definition_list')},
+        {'text': quota_attribute_definition.name, 'url': ""},
+    ]
+    context = {'form': form, 'quota_attribute_definition': quota_attribute_definition,
+               'object_name': "quota_attribute_definition", 'breadcrumbs': breadcrumbs, 'action': "edit"}
+    return render(request, 'generics/generic_form.html', context)
+
+
+@user_passes_test(lambda u: u.is_superuser)
+def quota_attribute_definition_create(request):
+    if request.method == 'POST':
+        form = QuotaAttributeDefinitionForm(request.POST)
+        if form.is_valid():
+            form.save()
+            return redirect("profiles:quota_attribute_definition_list")
+    else:
+        form = QuotaAttributeDefinitionForm()
+    breadcrumbs = [
+        {'text': 'Quota attributes', 'url': reverse('profiles:quota_attribute_definition_list')},
+        {'text': 'Create a new quota attribute', 'url': ""},
+    ]
+    context = {'form': form, 'object_name': "quota_attribute_definition", 'breadcrumbs': breadcrumbs,
+               'action': "create"}
+    return render(request, 'generics/generic_form.html', context)
+
+
+@user_passes_test(lambda u: u.is_superuser)
+def quota_attribute_definition_delete(request, quota_attribute_definition_id):
+    quota_attribute_definition = get_object_or_404(QuotaAttributeDefinition, id=quota_attribute_definition_id)
+    if request.method == 'POST':
+        quota_attribute_definition.delete()
+        return redirect("profiles:quota_attribute_definition_list")
+    args = {
+        "quota_attribute_definition_id": quota_attribute_definition_id,
+    }
+    breadcrumbs = [
+        {'text': 'Quota attributes', 'url': reverse('profiles:quota_attribute_definition_list')},
+        {'text': quota_attribute_definition.name, 'url': ""}
+    ]
+    context = {
+        'breadcrumbs': breadcrumbs,
+        'confirm_text': mark_safe(f"Confirm deletion of <strong>{quota_attribute_definition.name}</strong>?"),
+        'action_url': reverse('profiles:quota_attribute_definition_delete', kwargs=args),
+        'button_text': 'Delete',
+        'details': {'warning_sentence': 'Warning: some users are still present in this quota_attribute_definition:',
```

Comment:
Not "users" but attribute definition

Reply:
This quota attribute was linked to the following billing groups (list of billing group)